### PR TITLE
8285802: AArch64: Consistently handle offsets in MacroAssembler as 64-bit quantities

### PR DESCRIPTION
**Pull request by theRealAph**
In a few places we pass offsets as 32-bit ints rather than 64-bit longs, and we're not consistent about it. We shold have a clean 64-bit path for offset generation and validation.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8285802](https://bugs.openjdk.java.net/browse/JDK-8285802): AArch64: Consistently handle offsets in MacroAssembler as 64-bit quantities

### Reviewers
 * [Nick Gasson](https://openjdk.java.net/census#ngasson) (@nick-arm - **Reviewer**)
 * [Andrew Dinn](https://openjdk.java.net/census#adinn) (@adinn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8448/head:pull/8448` \
`$ git checkout pull/8448`

Update a local copy of the PR: \
`$ git checkout pull/8448` \
`$ git pull https://git.openjdk.java.net/jdk pull/8448/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8448`

View PR using the GUI difftool: \
`$ git pr show -t 8448`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8448.diff">https://git.openjdk.java.net/jdk/pull/8448.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp [hunk 1]
```diff
@@ -6521,7 +6521,7 @@ class StubGenerator: public StubCodeGenerator {
     assert(is_even(framesize/2), "sp not 16-byte aligned");
 
     // lr and fp are already in place
-    __ sub(sp, rfp, ((unsigned)framesize-4) << LogBytesPerInt); // prolog
+    __ sub(sp, rfp, ((uint64_t)framesize-4) << LogBytesPerInt); // prolog
```
Comment (nick-arm) [inline src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp:6524]: Typo prolog -> prologue in the existing code.

**Review by nick-arm — APPROVED** [review]
Seems fine. Was this fixing a particular issue?

**Comment by theRealAph** [comment]
> Seems fine. Was this fixing a particular issue?

It's a latent bug. I don't think it manifests today, but it's a bug.

File: src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp (hunk 1 above)
Comment (theRealAph) [reply src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp:6524]: I think that "prolog" is a correct variant of "prologue" in American English.

**Review by adinn — APPROVED** [review]
Looks ok, even on Windows! (the UI test that failed was because of a timeout so I assume it can be disregarded).

**Comment by vnkozlov** [comment]
These changes introduced regression: https://bugs.openjdk.java.net/browse/JDK-8285923

**Pull request closed without merging**